### PR TITLE
support gloo monorepo

### PR DESCRIPTION
**Pull request by axhixh**
changed thetool to work with new monorepo

### CODE REVIEW

File: cmd/addon/addon.go [hunk 1]
```diff
@@ -20,45 +20,31 @@ type configurator interface {
 }
 type Addon struct {
 	Name          string                 `json:"name"`
-	Repository    string                 `json:"repository,omitempty"`
-	Commit        string                 `json:"commit,omitempty"`
-	Image         string                 `json:"dockerImage,omitempty"`
-	Tag           string                 `json:"dockerTag,omitempty"`
-	Enable        bool                   `json:"enable"`
-	Configuration map[string]interface{} `json:"configuration,omitempty"`
+	Configuration map[string]interface{} `json:"configuration"`
```
Comment (yuval-k) [inline cmd/addon/addon.go:28]: are all the removed properties optional? for example, is enabled not common to every addon? and if so should it be here?

File: cmd/addon/addon.go [hunk 2]
```diff
@@ -83,32 +68,24 @@ func (s *Addon) SafeName() string {
...
-	if s.Tag != "" {
-		return s.Tag
-	}
-	if len(s.Commit) >= 7 {
-		return s.Commit[:7]
-	}
-	return ""
+
+func (s *Addon) IsGlooAddon() bool {
+	return s.Configuration != nil && s.Configuration["gloo"] == true
```
Comment (yuval-k) [inline cmd/addon/addon.go]: "gloo" is more than once; perhaps const?

File: cmd/addon/addon.go [hunk 3]
```diff
@@ -83,32 +68,24 @@ func (s *Addon) SafeName() string {
...
-	}
-	fmt.Fprintf(b, "%-12s: %v\n", "Enable", s.Enable)
-	if s.Enable && s.Configuration != nil {
+	if s.Configuration != nil {
 		status, ok := s.Configuration["status"]
 		if ok {
 			fmt.Fprintf(b, "%-12s: %v\n", "Status", status)
 		}
+
+		enabled, ok := s.Configuration["enable"]
```
Comment (yuval-k) [inline cmd/addon/addon.go]: "enabled" used more than once. const?

File: cmd/addon/configurator.go [hunk 4]
```diff
@@ -12,11 +12,16 @@ type EnableDisable struct {
...
-	if a.Enable {
-		defaultSelection = "enable"
-	} else {
-		defaultSelection = "disable"
+	defaultSelection := "enable"
+	if a.Configuration != nil {
+		v, ok := a.Configuration["enable"]
+		if ok {
+			if v.(bool) {
+				defaultSelection = "enable"
```
Comment (yuval-k) [inline cmd/addon/configurator.go]: const?

File: cmd/addon/addon.go (hunk 1 above)
Comment (axhixh) [reply cmd/addon/addon.go:28]: enabled is only present for "gloo" addons and other addons have "status". Could use "status" for everything, but didn't do it to make Helm chart templates easier.

**Review by yuval-k — APPROVED** [review]

**Pull request merged**